File: custom-addons/morenitapp/models/cofradia.py
from odoo import models, fields, api

# Modelo Cofradia
# Guarda información de cofradías

class Cofradia(models.Model):
    _name = 'morenitapp.cofradia'
    _description = 'Cofradía'

    # Campo representativo
    _rec_name = 'nombreCofradia'


    # CIF cofradía
    cifCofradia = fields.Char(
        string="CIF de Cofradía",
        required=True
    )

    # Nombre cofradía
    nombreCofradia = fields.Char(
        string="Nombre de Cofradía",
        required=True
    )

    # Año fundación
    antiguedadCofradia = fields.Integer(
        string="Año de Fundación"
    )

    # Responsable
    nombreResponsable = fields.Char(
        string="Nombre del Responsable"
    )

    calle_id = fields.Many2one(
        'morenitapp.calle',
        string="Calle"
    )

    # Campos relacionados automáticos
    localidad_id = fields.Many2one(
        'morenitapp.localidad',
        related='calle_id.localidad_id',
        string="Localidad",
        store=True,
        readonly=True
    )

    codPostal_id = fields.Many2one(
        'morenitapp.codigopostal',
        related='calle_id.codPostal_id',
        string="C.P.",
        store=True,
        readonly=True
    )

    provincia_id = fields.Many2one(
        'morenitapp.provincia',
        related='localidad_id.codProvincia_id',
        string="Provincia",
        store=True,
        readonly=True
    )

    # Datos adicionales dirección
    puerta = fields.Char(string="Puerta")
    piso = fields.Char(string="Piso")
    numero = fields.Char(string="Número")
    bloque = fields.Char(string="Bloque")
    escalera = fields.Char(string="Escalera")
    portal = fields.Char(string="Portal")

    telefonoCofradia = fields.Char(
        string="Teléfono"
    )

    emailCofradia = fields.Char(
        string="Email"
    )

    paginaWeb = fields.Char(
        string="Página Web"
    )

    # Observaciones
    observaciones = fields.Text(
        string="Observaciones"
    )

File: mobile_flutter/morenitapp/lib/features/panel-gestion/secretaria/domain/entities/cofradia.dart
class Cofradia {
  final String id;
  final String cif;
  final String nombre;
  final int fundacion;
  final String email;
  final String telefono;
  final String web;
  final String observaciones;

  final int? calleId;
  final String calleNombre;
  final String numero;
  final String piso;
  final String puerta;
  final String bloque;
  final String escalera;
  final String portal;

  Cofradia({
    required this.id,
    required this.cif,
    required this.nombre,
    required this.fundacion,
    required this.email,
    required this.telefono,
    required this.web,
    required this.observaciones,
    required this.calleId,
    required this.calleNombre,
    required this.numero,
    required this.piso,
    required this.puerta,
    required this.bloque,
    required this.escalera,
    required this.portal,
  });

  static String clean(v) => (v == null || v == "" || v == false) ? "" : v.toString();

  factory Cofradia.fromJson(Map<String, dynamic> json) {
    final dir = json["direccion"] ?? {};

    return Cofradia(
      id: clean(json["id"]),
      cif: clean(json["cifCofradia"]),
      nombre: clean(json["nombreCofradia"]),
      fundacion: json["antiguedadCofradia"] is int ? json["antiguedadCofradia"] : 0,
      email: clean(json["emailCofradia"]),
      telefono: clean(json["telefonoCofradia"]),
      web: clean(json["paginaWeb"]),
      observaciones: clean(json["observaciones"]),

      calleId: dir["calle_id"],
      calleNombre: clean(dir["calle_name"]),
      numero: clean(dir["numero"]),
      piso: clean(dir["piso"]),
      puerta: clean(dir["puerta"]),
      bloque: clean(dir["bloque"]),
      escalera: clean(dir["escalera"]),
      portal: clean(dir["portal"]),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "cifCofradia": cif,
      "nombreCofradia": nombre,
      "antiguedadCofradia": fundacion,
      "emailCofradia": email,
      "telefonoCofradia": telefono,
      "paginaWeb": web,
      "observaciones": observaciones,

      "calle_id": calleId,
      "numero": numero,
      "piso": piso,
      "puerta": puerta,
      "bloque": bloque,
      "escalera": escalera,
      "portal": portal,
    };
  }
}
